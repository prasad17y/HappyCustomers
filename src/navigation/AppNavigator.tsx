import React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import UserListScreen from '../screens/UserListScreen';
import AddUserScreen from '../screens/AddUserScreen';
import {RootStackParamList} from './types';
import {Screens} from './routes';

const RootStack = createNativeStackNavigator<RootStackParamList>();

const AppNavigator = () => {
  return (
    <RootStack.Navigator>
      <RootStack.Group screenOptions={{headerShown: false}}>
        <RootStack.Screen name={Screens.UserList} component={UserListScreen} />
      </RootStack.Group>
      <RootStack.Group
        screenOptions={{
          presentation: 'fullScreenModal',
          headerShown: false,
        }}>
        <RootStack.Screen name={Screens.AddUser} component={AddUserScreen} />
      </RootStack.Group>
    </RootStack.Navigator>
  );
};

export default AppNavigator;
