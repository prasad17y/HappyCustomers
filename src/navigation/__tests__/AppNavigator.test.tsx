import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';
import AppNavigator from '../AppNavigator';
// import {Text, TouchableOpacity} from 'react-native';

// Mock all the screen components to be simple placeholders
jest.mock('../../screens/UserListScreen', () => {
  const React = require('react');
  const {Text, TouchableOpacity} = require('react-native');
  const {useNavigation} = require('@react-navigation/native');
  return () => {
    const navigation = useNavigation();
    return (
      <>
        <Text>User List Screen</Text>
        <TouchableOpacity onPress={() => navigation.navigate('AddUser')}>
          <Text>Go to Add User</Text>
        </TouchableOpacity>
      </>
    );
  };
});
jest.mock('../../screens/AddUserScreen', () => {
  const React = require('react');
  const {Text} = require('react-native');
  return () => <Text>Add User Screen</Text>;
});
jest.mock('../../screens/EditUserScreen', () => {
  const React = require('react');
  const {Text} = require('react-native');
  return () => <Text>Edit User Screen</Text>;
});

describe('AppNavigator', () => {
  it('should render the initial UserList screen', () => {
    render(
      <NavigationContainer>
        <AppNavigator />
      </NavigationContainer>,
    );

    // Verify that the text from our mocked UserListScreen is visible
    expect(screen.getByText('User List Screen')).toBeTruthy();
  });

  it('should navigate to the AddUser screen when triggered', () => {
    render(
      <NavigationContainer>
        <AppNavigator />
      </NavigationContainer>,
    );

    // Ensure the initial screen is visible
    expect(screen.getByText('User List Screen')).toBeTruthy();

    // Simulate a press on the button inside our mocked UserListScreen
    fireEvent.press(screen.getByText('Go to Add User'));

    // Verify that the app has navigated and the AddUserScreen is now visible
    expect(screen.getByText('Add User Screen')).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = render(
      <NavigationContainer>
        <AppNavigator />
      </NavigationContainer>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
