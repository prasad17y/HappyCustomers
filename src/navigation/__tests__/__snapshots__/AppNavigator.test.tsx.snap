// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigator should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <Text>
            User List Screen
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text>
              Go to Add User
            </Text>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="UserList"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
